platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/ConfigurationInstaller.java

```diff
@@ -0,0 +1,168 @@
+
+/**
+ * Replace the Felix {@code ConfigInstaller} with our own implementation that properly syncs the etc
+ * directory using the {@code felix.fileinstall.filename} property. Handling a config delete is fairly
+ * straight forward. The following outcomes are possible when an update occurs:
+ * <p>
+ * <table>
+ * <tr>
+ * <td>felixFileOriginal</td><td>felixFileNew</td><td>comments</td>
+ * </tr>
+ * <tr>
+ * <td>null</td><td>null</td><td>Felix file prop wasn't tracked and is still not tracked</td>
+ * </tr>
+ * <tr>
+ * <td>null</td><td>X</td><td>New config with a new Felix file prop to track</td>
+ * </tr>
+ * <tr>
+ * <td>Y</td><td>null</td><td>Felix file prop no longer being tracked, delete config file and remove from map</td>
+ * </tr>
+ * <tr>
+ * <td>Y</td><td>X</td><td>Felix file prop updated, delete config file and update map</td>
+ * </tr>
+ * <tr>
+ * <td>Y</td><td>Y</td><td>Felix file prop did not change, do nothing</td>
+ * </tr>
+ * </table>
+ *
+ * @see #handleUpdate(FelixConfig)
+ * @see #handleDelete(String)
+ */
+public class ConfigurationInstaller implements SynchronousConfigurationListener {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfigurationInstaller.class);
+
+    private final ConfigurationAdmin configurationAdmin;
+
+    private Map<String, File> pidFileMap;
+
+    public ConfigurationInstaller(ConfigurationAdmin configurationAdmin) {
+        this.configurationAdmin = configurationAdmin;
+        this.pidFileMap = Collections.emptyMap();
+    }
+
+    /**
+     * @return a read-only map of configuration pids to felix file install properties.
+     */
+    public Map<String, File> getPidFileMap() {
+        return Collections.unmodifiableMap(pidFileMap);
+    }
+
+    public void init() throws IOException, InvalidSyntaxException {
+        this.pidFileMap = Arrays.stream(configurationAdmin.listConfigurations(null))
+                .map(FelixConfig::new)
+                .filter(config -> config.getFile() != null)
+                .collect(Collectors.toMap(FelixConfig::getPid, FelixConfig::getFile,
+                        // Use the latest value for the same key
+                        (pid1, pid2) -> pid2));
+    }
+
+    @Override
+    public void configurationEvent(ConfigurationEvent configurationEvent) {
+        try {
+            switch (configurationEvent.getType()) {
+            case CM_UPDATED:
+                LOGGER.debug("Handling update");
+                handleUpdate(new FelixConfig(() -> getConfiguration(configurationEvent)));
+                break;
+            case CM_DELETED:
+                LOGGER.debug("Handling delete");
+                handleDelete(configurationEvent.getPid());
+                break;
+            default:
+                LOGGER.debug("Unknown configuration event type, taking no action");
+            }
+        } catch (IOException e) {
+            LOGGER.debug("Error reading or modifying configurations on disk: {}", e);
```

Comment:
The fact is that if an IOException occurs, the file was not properly saved back to etc which means the configuration will be lost the next time the system restarts. This is important enough to log at something higher than debug. The error should not be propagated up to Felix.